github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/commitchecks/impl/GitHubBranchCommitMessageCheck.java

```diff
@@ -39,23 +45,23 @@ public GitHubBranchCommitMessageCheck() {
     }
 
     @Override
-    public GitHubBranchCause check(GHBranch remoteBranch, GitHubBranchRepository localRepo, Commit[] commits) {
-        String name = remoteBranch.getName();
-        if (matchCriteria.isEmpty()) {
-            LOG.warn("Commit message event added but no match criteria set, all commits are allowed.");
-            return null;
-        }
-
-        List<String> messages = Stream.of(commits)
-                .map(commit -> commit.getCommit().getMessage())
-                .collect(Collectors.toList());
-
-        if (commitsAreAllowed(messages)) {
-            LOG.debug("Commit messages {} for branch [{}] allowed, commit ignored.", messages, name);
+    public GitHubBranchCause check(GHBranch remoteBranch, GitHubBranchRepository localRepo, GHCommit commit) {
+        try {
+            List<String> messages = Arrays.asList(commit.getCommitShortInfo().getMessage());
+            return check(remoteBranch, localRepo, () -> messages);
+        } catch (IOException e) {
+            LOG.error("Failed to get commit message for hash [{}]", commit.getSHA1(), e);
```

Comment:
questionable to catch IOExceptions, that doesn't allow trigger to handle errors on top level. Code can't be sure whether it safe to allow run build or not. Trigger may have multiple events or grouping events, imho IOException should fail trigger run and ideally skip it. Skipping instead triggering is safer. Or it may end with such fun as https://github.com/jenkinsci/branch-api-plugin/issues/85